### ISSUE TITLE
Attributes are not mapped to the claim with RestApiClaimProvider 

### ISSUE THREAD

**Issue by twobeeb** [tchiotludo/akhq#957]
I'm observing a very strange behavior in ``dev`` branch. It's working fine in 0.18 (didn't test 0.19)

The JWTClaimSet is properly generated during authentication.
````
2021-12-16 10:49:40,691 DEBUG  35m-thread-250 .g.c.JWTClaimsSetGenerator  Generated claim set: 
  {
    sub=f297755, nbf=1639651780, 
    connectsFilterRegexp=List(.*), 
    roles=[topic/read, topic/data/read, group/read, registry/read, connect/read, connect/state/update], 
    iss=akhq, exp=1639655380,         
    iat=1639651780, 
    consumerGroupsFilterRegexp=List(.*), 
    topicsFilterRegexp=List(.*)
}
````
But then when it's converted to JWT and parsed during the following calls, its attributes are not mapped: 
````
2021-12-16 10:49:40,849 DEBUG  35mpGroup-1-30 s.t.r.DefaultTokenResolver  Token eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmMjk3NzU1IiwibmJmIjoxNjM5NjUxNzgwLCJjb25uZWN0c0ZpbHRlclJlZ2V4cCI6e30sInJvbGVzIjpbInRvcGljXC9yZWFkIiwidG9waWNcL2RhdGFcL3JlYWQiLCJncm91cFwvcmVhZCIsInJlZ2lzdHJ5XC9yZWFkIiwiY29ubmVjdFwvcmVhZCIsImNvbm5lY3RcL3N0YXRlXC91cGRhdGUiXSwiaXNzIjoiYWtocSIsImV4cCI6MTYzOTY1NTM4MCwiaWF0IjoxNjM5NjUxNzgwLCJjb25zdW1lckdyb3Vwc0ZpbHRlclJlZ2V4cCI6e30sInRvcGljc0ZpbHRlclJlZ2V4cCI6e319.MEbKPCUpW6LQVBV2LxCGNgvvSxiwo98b-9Cbvy3OT8g found in request GET /linked/api/cluster
2021-12-16 10:49:40,849 DEBUG  35mpGroup-1-30 i.m.s.t.j.v.JwtValidator    Validating signed JWT
2021-12-16 10:49:40,850 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      
Attributes: 
    sub=>f297755, nbf=>Thu Dec 16 10:49:40 UTC 2021, 
    connectsFilterRegexp=>{}, 
    roles=>["topic\/read","topic\/data\/read","group\/read","registry\/read","connect\/read","connect\/state\/update"], 
    iss=>akhq, 
    exp=>Thu Dec 16 11:49:40 UTC 2021, 
    iat=>Thu Dec 16 10:49:40 UTC 2021, 
    consumerGroupsFilterRegexp=>{}, 
    topicsFilterRegexp=>{}
````
This is particularly important because the value ``{ }`` is interpreted as "full access" in AKHQ. So instead of not seeing any topics, the users see all topics.

The complete authentication logs followed by a
````
2021-12-16 10:49:40,457 DEBUG  35mpGroup-1-30 s.t.r.DefaultTokenResolver  Request POST, /linked/login, no token found.
2021-12-16 10:49:40,457 DEBUG  35mpGroup-1-30 .m.s.r.InterceptUrlMapRule  One or more of the IP patterns matched the host address [198.18.135.218]. Continuing request processing.
2021-12-16 10:49:40,458 DEBUG  35mpGroup-1-30 m.s.r.AbstractSecurityRule  The given roles [[isAnonymous()]] matched one or more of the required roles [[isAnonymous()]]. Allowing the request
2021-12-16 10:49:40,458 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      Authorized request POST /linked/login. The rule provider org.akhq.modules.SecuredAnnotationRuleWithDefault authorized the request.
2021-12-16 10:49:40,461 DEBUG  35mpGroup-1-30 i.m.s.a.Authenticator       org.akhq.modules.BasicAuthAuthenticationProviderio.micronaut.security.ldap.LdapAuthenticationProvider
2021-12-16 10:49:40,462 DEBUG  35m-thread-250 LdapAuthenticationProvider  Starting authentication with configuration [default]
2021-12-16 10:49:40,462 DEBUG  35m-thread-250 LdapAuthenticationProvider  Attempting to initialize manager context
2021-12-16 10:49:40,463 DEBUG  35m-thread-250 LdapAuthenticationProvider  Manager context initialized successfully
2021-12-16 10:49:40,463 DEBUG  35m-thread-250 LdapAuthenticationProvider  Attempting to authenticate with user [f297755]
2021-12-16 10:49:40,510 DEBUG  35m-thread-250 LdapAuthenticationProvider  User found in context [cn=F297755,ou=FR,ou=ZONE3,o=MICHELIN,c=FR]. Attempting to bind.
2021-12-16 10:49:40,593 DEBUG  35m-thread-250 LdapAuthenticationProvider  Successfully bound user [cn=F297755,ou=FR,ou=ZONE3,o=MICHELIN,c=FR]. Attempting to retrieving groups.
2021-12-16 10:49:40,648 TRACE  35m-thread-250 .DefaultLdapGroupProcessor  The following groups were found for [cn=F297755,ou=FR,ou=ZONE3,o=MICHELIN,c=FR]: [GP-MICHELIN-F4M-DEVOPS]
2021-12-16 10:49:40,648 DEBUG  35m-thread-250 .DefaultLdapGroupProcessor  Attempting to add any additional groups...
2021-12-16 10:49:40,648 DEBUG  35m-thread-250 LdapAuthenticationProvider  Group search returned [[GP-MICHELIN-F4M-DEVOPS]] for user [f297755]
2021-12-16 10:49:40,648 TRACE  35m-thread-250 LdapAuthenticationProvider  Attempting to map [f297755] with groups [[GP-MICHELIN-F4M-DEVOPS]] to an authentication response.
2021-12-16 10:49:40,650 TRACE  35m-thread-250 redentialsHttpClientFilter  Did not find any OAuth 2.0 client which should decorate the request with an access token received from client credentials request
2021-12-16 10:49:40,682 DEBUG  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Sending HTTP POST to http://claim-api/akhq-claim
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Content-Type: application/json
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Accept: application/json
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  host: claim-api
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  connection: close
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  content-length: 114
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Request Body
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  ----
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  {"providerType":"LDAP","username":"f297755","groups":["GP-MICHELIN-F4M-DEVOPS"]}
2021-12-16 10:49:40,682 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  ----
2021-12-16 10:49:40,688 DEBUG  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Received response 200 from http://claim-api/akhq-claim
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Date: Thu, 16 Dec 2021 10:49:40 GMT
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Content-Type: application/json
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Connection: close
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  content-length: 266
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  Response Body
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  ----
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  {
  "roles" : [ "topic/read", "topic/data/read", "group/read", "registry/read", "connect/read", "connect/state/update" ],
  "attributes" : {
    "consumerGroupsFilterRegexp" : [ ".*" ],
    "connectsFilterRegexp" : [ ".*" ],
    "topicsFilterRegexp" : [ ".*" ]
  }
}
2021-12-16 10:49:40,688 TRACE  35mpGroup-1-40 .m.h.c.n.DefaultHttpClient  ----
2021-12-16 10:49:40,690 DEBUG  35m-thread-250 ccessRefreshTokenGenerator  Skipped refresh token generation because no io.micronaut.security.token.validator.RefreshTokenValidator implementation is present
2021-12-16 10:49:40,690 DEBUG  35m-thread-250 .g.c.JWTClaimsSetGenerator  Setting expiration to 3600
2021-12-16 10:49:40,691 DEBUG  35m-thread-250 .g.c.JWTClaimsSetGenerator  Generated claim set: {sub=f297755, nbf=1639651780, connectsFilterRegexp=List(.*), roles=[topic/read, topic/data/read, group/read, registry/read, connect/read, connect/state/update], iss=akhq, exp=1639655380, iat=1639651780, consumerGroupsFilterRegexp=List(.*), topicsFilterRegexp=List(.*)}
2021-12-16 10:49:40,691 INFO   35m-thread-250 org.akhq.log.access         [Date: 2021-12-16T10:49:40.458228Z] [Duration: 233 ms] [Url: POST /linked/login] [Status: 303] [Ip: /198.18.135.218] [User: Anonymous]
2021-12-16 10:49:40,693 DEBUG  35m-thread-250 LdapAuthenticationProvider  Response successfully created for [f297755]. Response is authenticated: [true]

# End of authentication phase

# Following GET on /api/me

2021-12-16 10:49:40,750 DEBUG  35mpGroup-1-30 .t.r.HttpHeaderTokenReader  Looking for bearer token in Authorization header
2021-12-16 10:49:40,750 DEBUG  35mpGroup-1-30 s.t.r.DefaultTokenResolver  Token eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmMjk3NzU1IiwibmJmIjoxNjM5NjUxNzgwLCJjb25uZWN0c0ZpbHRlclJlZ2V4cCI6e30sInJvbGVzIjpbInRvcGljXC9yZWFkIiwidG9waWNcL2RhdGFcL3JlYWQiLCJncm91cFwvcmVhZCIsInJlZ2lzdHJ5XC9yZWFkIiwiY29ubmVjdFwvcmVhZCIsImNvbm5lY3RcL3N0YXRlXC91cGRhdGUiXSwiaXNzIjoiYWtocSIsImV4cCI6MTYzOTY1NTM4MCwiaWF0IjoxNjM5NjUxNzgwLCJjb25zdW1lckdyb3Vwc0ZpbHRlclJlZ2V4cCI6e30sInRvcGljc0ZpbHRlclJlZ2V4cCI6e319.MEbKPCUpW6LQVBV2LxCGNgvvSxiwo98b-9Cbvy3OT8g found in request GET /linked/api/me
2021-12-16 10:49:40,750 DEBUG  35mpGroup-1-30 i.m.s.t.j.v.JwtValidator    Validating signed JWT
2021-12-16 10:49:40,751 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      Attributes: sub=>f297755, nbf=>Thu Dec 16 10:49:40 UTC 2021, connectsFilterRegexp=>{}, roles=>["topic\/read","topic\/data\/read","group\/read","registry\/read","connect\/read","connect\/state\/update"], iss=>akhq, exp=>Thu Dec 16 11:49:40 UTC 2021, iat=>Thu Dec 16 10:49:40 UTC 2021, consumerGroupsFilterRegexp=>{}, topicsFilterRegexp=>{}
2021-12-16 10:49:40,751 DEBUG  35mpGroup-1-30 .m.s.r.InterceptUrlMapRule  One or more of the IP patterns matched the host address [198.18.135.218]. Continuing request processing.
2021-12-16 10:49:40,751 DEBUG  35mpGroup-1-30 m.s.r.AbstractSecurityRule  The given roles [[topic/read, topic/data/read, group/read, registry/read, connect/read, connect/state/update, isAnonymous(), isAuthenticated()]] matched one or more of the required roles [[isAnonymous()]]. Allowing the request
2021-12-16 10:49:40,751 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      Authorized request GET /linked/api/me. The rule provider org.akhq.modules.SecuredAnnotationRuleWithDefault authorized the request.
2021-12-16 10:49:40,755 INFO   35m-thread-250 org.akhq.log.access         [Date: 2021-12-16T10:49:40.751861Z] [Duration: 3 ms] [Url: GET /linked/api/me] [Status: 200] [Ip: /198.18.135.218] [User: f297755]
2021-12-16 10:49:40,849 DEBUG  35mpGroup-1-30 .t.r.HttpHeaderTokenReader  Looking for bearer token in Authorization header
2021-12-16 10:49:40,849 DEBUG  35mpGroup-1-30 s.t.r.DefaultTokenResolver  Token eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmMjk3NzU1IiwibmJmIjoxNjM5NjUxNzgwLCJjb25uZWN0c0ZpbHRlclJlZ2V4cCI6e30sInJvbGVzIjpbInRvcGljXC9yZWFkIiwidG9waWNcL2RhdGFcL3JlYWQiLCJncm91cFwvcmVhZCIsInJlZ2lzdHJ5XC9yZWFkIiwiY29ubmVjdFwvcmVhZCIsImNvbm5lY3RcL3N0YXRlXC91cGRhdGUiXSwiaXNzIjoiYWtocSIsImV4cCI6MTYzOTY1NTM4MCwiaWF0IjoxNjM5NjUxNzgwLCJjb25zdW1lckdyb3Vwc0ZpbHRlclJlZ2V4cCI6e30sInRvcGljc0ZpbHRlclJlZ2V4cCI6e319.MEbKPCUpW6LQVBV2LxCGNgvvSxiwo98b-9Cbvy3OT8g found in request GET /linked/api/cluster
2021-12-16 10:49:40,849 DEBUG  35mpGroup-1-30 i.m.s.t.j.v.JwtValidator    Validating signed JWT
2021-12-16 10:49:40,850 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      Attributes: sub=>f297755, nbf=>Thu Dec 16 10:49:40 UTC 2021, connectsFilterRegexp=>{}, roles=>["topic\/read","topic\/data\/read","group\/read","registry\/read","connect\/read","connect\/state\/update"], iss=>akhq, exp=>Thu Dec 16 11:49:40 UTC 2021, iat=>Thu Dec 16 10:49:40 UTC 2021, consumerGroupsFilterRegexp=>{}, topicsFilterRegexp=>{}
2021-12-16 10:49:40,850 DEBUG  35mpGroup-1-30 .m.s.r.InterceptUrlMapRule  One or more of the IP patterns matched the host address [198.18.135.218]. Continuing request processing.
2021-12-16 10:49:40,851 DEBUG  35mpGroup-1-30 i.m.s.f.SecurityFilter      Authorized request GET /linked/api/cluster. The rule provider org.akhq.modules.SecuredAnnotationRuleWithDefault authorized the request.

````

**Comment by twobeeb** [comment]
I don't have this behavior when using the LocalSecurityClaimProvider
````
2021-12-16 12:28:40,800 DEBUG thread-113 .g.c.JWTClaimsSetGenerator Generated claim set: {sub=f297755, nbf=1639657720, connectsFilterRegexp=[.*], roles=[topic/read, topic/data/read, node/read, group/read, registry/read, acls/read, connect/read, connect/state/update], iss=akhq, exp=1639661320, iat=1639657720, consumerGroupsFilterRegexp=[], topicsFilterRegexp=[.*]}
2021-12-16 12:28:40,801 INFO  thread-113 org.akhq.log.access        [Date: 2021-12-16T12:28:40.553826Z] [Duration: 247 ms] [Url: POST /old/login] [Status: 303] [Ip: /198.18.139.127] [User: Anonymous]
2021-12-16 12:28:40,803 DEBUG thread-113 LdapAuthenticationProvider Response successfully created for [f297755]. Response is authenticated: [true]
2021-12-16 12:28:40,856 DEBUG Group-1-3 .t.r.HttpHeaderTokenReader  Looking for bearer token in Authorization header
2021-12-16 12:28:40,856 DEBUG Group-1-3 s.t.r.DefaultTokenResolver  Token eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmMjk3NzU1IiwibmJmIjoxNjM5NjU3NzIwLCJjb25uZWN0c0ZpbHRlclJlZ2V4cCI6WyIuKiJdLCJyb2xlcyI6WyJ0b3BpY1wvcmVhZCIsInRvcGljXC9kYXRhXC9yZWFkIiwibm9kZVwvcmVhZCIsImdyb3VwXC9yZWFkIiwicmVnaXN0cnlcL3JlYWQiLCJhY2xzXC9yZWFkIiwiY29ubmVjdFwvcmVhZCIsImNvbm5lY3RcL3N0YXRlXC91cGRhdGUiXSwiaXNzIjoiYWtocSIsImV4cCI6MTYzOTY2MTMyMCwiaWF0IjoxNjM5NjU3NzIwLCJjb25zdW1lckdyb3Vwc0ZpbHRlclJlZ2V4cCI6W10sInRvcGljc0ZpbHRlclJlZ2V4cCI6WyIuKiJdfQ.Q9QBjmpY2_OZtgXWxS3Rddw7Ceep_BaWOePT0TIoTwo found in request GET /old/api/me
2021-12-16 12:28:40,857 DEBUG Group-1-3 i.m.s.t.j.v.JwtValidator    Validating signed JWT
2021-12-16 12:28:40,869 DEBUG Group-1-3 i.m.s.f.SecurityFilter      Attributes: sub=>f297755, nbf=>Thu Dec 16 12:28:40 UTC 2021, connectsFilterRegexp=>[".*"], roles=>["topic\/read","topic\/data\/read","node\/read","group\/read","registry\/read","acls\/read","connect\/read","connect\/state\/update"], iss=>akhq, exp=>Thu Dec 16 13:28:40 UTC 2021, iat=>Thu Dec 16 12:28:40 UTC 2021, consumerGroupsFilterRegexp=>[], topicsFilterRegexp=>[".*"]
2021-12-16 12:28:40,870 DEBUG Group-1-3 .m.s.r.InterceptUrlMapRule  One or more of the IP patterns matched the host address [198.18.139.127]. Continuing request processing.
2021-12-16 12:28:40,870 DEBUG Group-1-3 m.s.r.AbstractSecurityRule  The given roles [[topic/read, topic/data/read, node/read, group/read, registry/read, acls/read, connect/read, connect/state/update, isAnonymous(), isAuthenticated()]] matched one or more of the required roles [[isAnonymous()]]. Allowing the request
2021-12-16 12:28:40,870 DEBUG Group-1-3 i.m.s.f.SecurityFilter      Authorized request GET /old/api/me. The rule provider org.akhq.modules.SecuredAnnotationRuleWithDefault authorized the request.
2021-12-16 12:28:40,873 INFO  thread-113 org.akhq.log.access        [Date: 2021-12-16T12:28:40.871077Z] [Duration: 1 ms] [Url: GET /old/api/me] [Status: 200] [Ip: /198.18.139.127] [User: f297755]
````

Notice the difference between RestApiClaimProvider and LocalSecurityClaimProvider :
````
# RestApiClaimProvider
Generated claim set: {sub=f297755, nbf=1639651780, connectsFilterRegexp=List(.*), roles=[topic/read, topic/data/read, group/read, registry/read, connect/read, connect/state/update], iss=akhq, exp=1639655380, iat=1639651780, consumerGroupsFilterRegexp=List(.*), topicsFilterRegexp=List(.*)}

# LocalSecurityClaimProvider 
Generated claim set: {sub=f297755, nbf=1639657720, connectsFilterRegexp=[.*], roles=[topic/read, topic/data/read, node/read, group/read, registry/read, acls/read, connect/read, connect/state/update], iss=akhq, exp=1639661320, iat=1639657720, consumerGroupsFilterRegexp=[], topicsFilterRegexp=[.*]}


````

**Comment by tchiotludo** [comment]
I try quickly, LocalSecurityClaimProvider works well and I have unit test that validate everything works. 

I just see that isn't unit test on this `RestApiClaimProvider` so there is no validation that this is working. 
Maybe you can add some in order to find the issue ?

**Comment by twobeeb** [comment]
Agreed, I should mock a response to an external API to verify it produces the proper JWT token

**Comment by tchiotludo** [comment]
You can also used micronaut directly to respond what you need with mocking, even more simplest

**Comment by twobeeb** [comment]
So in summary:
1. Micronaut 3.x broke the deserialization of the Map\<String, Object\> used in AKHQClaimResponse
2. I added the necessary test to validate the failure on `dev` it in #959 
3. I have a proposition for a fix in #958, but maybe you have a better solution

This subject is not urgent since I'm currently using `0.18` in my environnements (I just temporarily deployed `dev` to validate #932). 
We can look at it in January.

Thanks
